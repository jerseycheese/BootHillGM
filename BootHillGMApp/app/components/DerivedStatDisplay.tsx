import React from 'react';

/**
 * DerivedStatDisplay component displays a derived stat with label, value, and optional description.
 */

interface DerivedStatDisplayProps {
  label: string; // The label of the derived stat (e.g., "Hit Points")
  value: number; // The current value of the derived stat
  description?: string; // A description of the derived stat (optional)
}

export default function DerivedStatDisplay({
  label,
  value,
  description,
}: DerivedStatDisplayProps) {
  return (
    <div className="stat-container flex flex-col gap-2">
      <div className="stat-label text-sm font-bold">{label}</div>
      <div className="stat-value text-lg">{value}</div>
      {description && (
        <div className="stat-description text-xs italic">{description}</div>
      )}
    </div>
  );
}
