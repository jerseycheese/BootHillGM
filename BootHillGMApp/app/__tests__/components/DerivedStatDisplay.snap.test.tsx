import React from 'react';
import { render } from '@testing-library/react';
import DerivedStatDisplay from '@/app/components/DerivedStatDisplay';

describe('DerivedStatDisplay snapshots', () => {
  it('matches snapshot with basic props', () => {
    const { container } = render(<DerivedStatDisplay label="Hit Points" value={24} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with description', () => {
    const { container } = render(
      <DerivedStatDisplay 
        label="Speed" 
        value={5} 
        description="Determines movement range and initiative"
      />
    );
    expect(container).toMatchSnapshot();
  });

  // Edge cases
  it('matches snapshot with zero value', () => {
    const { container } = render(<DerivedStatDisplay label="Bonus" value={0} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with large value', () => {
    const { container } = render(<DerivedStatDisplay label="Damage Potential" value={999} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with negative value', () => {
    const { container } = render(<DerivedStatDisplay label="Modifier" value={-3} />);
    expect(container).toMatchSnapshot();
  });

  // Testing specific elements
  it('renders with accessible attributes', () => {
    const { getByTestId } = render(
      <DerivedStatDisplay label="Accessibility" value={5} />
    );
    const derivedStatDisplay = getByTestId('derived-stat-display');
    expect(derivedStatDisplay).toBeInTheDocument();
    expect(derivedStatDisplay).toMatchSnapshot();
  });
});
