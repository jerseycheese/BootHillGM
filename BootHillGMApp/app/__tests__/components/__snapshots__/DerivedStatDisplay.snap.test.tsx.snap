// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DerivedStatDisplay snapshots matches snapshot with basic props 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-derived-stat-display"
    data-testid="derived-stat-display"
    id="bhgmDerivedStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Hit Points
    </div>
    <div
      class="stat-value text-lg"
    >
      24
    </div>
  </div>
</div>
`;

exports[`DerivedStatDisplay snapshots matches snapshot with description 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-derived-stat-display"
    data-testid="derived-stat-display"
    id="bhgmDerivedStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Speed
    </div>
    <div
      class="stat-value text-lg"
    >
      5
    </div>
    <div
      class="stat-description text-xs italic"
    >
      Determines movement range and initiative
    </div>
  </div>
</div>
`;

exports[`DerivedStatDisplay snapshots matches snapshot with large value 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-derived-stat-display"
    data-testid="derived-stat-display"
    id="bhgmDerivedStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Damage Potential
    </div>
    <div
      class="stat-value text-lg"
    >
      999
    </div>
  </div>
</div>
`;

exports[`DerivedStatDisplay snapshots matches snapshot with negative value 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-derived-stat-display"
    data-testid="derived-stat-display"
    id="bhgmDerivedStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Modifier
    </div>
    <div
      class="stat-value text-lg"
    >
      -3
    </div>
  </div>
</div>
`;

exports[`DerivedStatDisplay snapshots matches snapshot with zero value 1`] = `
<div>
  <div
    class="stat-container flex flex-col gap-2 bhgm-derived-stat-display"
    data-testid="derived-stat-display"
    id="bhgmDerivedStatDisplay"
  >
    <div
      class="stat-label text-sm font-bold"
    >
      Bonus
    </div>
    <div
      class="stat-value text-lg"
    >
      0
    </div>
  </div>
</div>
`;

exports[`DerivedStatDisplay snapshots renders with accessible attributes 1`] = `
<div
  class="stat-container flex flex-col gap-2 bhgm-derived-stat-display"
  data-testid="derived-stat-display"
  id="bhgmDerivedStatDisplay"
>
  <div
    class="stat-label text-sm font-bold"
  >
    Accessibility
  </div>
  <div
    class="stat-value text-lg"
  >
    5
  </div>
</div>
`;
